includes/class-wc-payment-tokens.php

```diff
@@ -0,0 +1,250 @@
+<?php
+
+if ( ! defined( 'ABSPATH' ) ) {
+	exit; // Exit if accessed directly
+}
+
+/**
+ * WooCommerce Payment Tokens
+ *
+ * An API for storing and managing tokens for gateways and customers.
+ *
+ * @class 		WC_Payment_Tokens
+ * @since		2.6.0
+ * @package		WooCommerce/Classes
+ * @category	Class
+ * @author		WooThemes
+ */
+class WC_Payment_Tokens {
+
+	/**
+	 * Makes sure that the passed data is suitable for storing a token
+	 * @param  array $args  An array of arguments (can be meta or core token field) to validate
+	 * @return boolean      True if the passed data is valid
+	 */
+	public static function validate( $args ) {
+		if ( empty( $args['token'] ) ) {
+			return false;
+		}
+
+		if ( empty( $args['type'] ) ) {
+			return false;
+		}
+
+		// If 'token' and 'type' are present, validate that any specific token types (credit card for example)
+		// contain their neccesary fields.
+		$is_valid = true;
+		$token_class = 'WC_Payment_Token_' . $args['type'];
+		if ( method_exists( $token_class, 'validate' ) ) {
```

Comment:
Been preferring is_callable of late, since that is the intent of this check. Worth swapping? Minor.

http://jmfeurprier.com/2010/01/03/method_exists-vs-is_callable/